### DISCUSSION TITLE
Lesson 6: Contract not found on mainnet-fork-dev

### DISCUSSION THREAD

**Discussion by carlitox477** [smartcontractkit/full-blockchain-solidity-course-py#308] [Q&A] [answered]
I create my alchemy project as Patrick said: 
* Enviroment: Deployment
* Chain: Ethereum
* Network: Mainnet

And i get the API key, which has the next structure: **https://eth-mainnet.alchemyapi.io/v2/API-KEY**

My brownie-config.yaml file is:
```yaml
dependencies:
  #- <organization/repo>@version
  - smartcontractkit/chainlink-brownie-contracts@1.1.1
compiler:
  solc:
    remappings:
      - '@chainlink=smartcontractkit/chainlink-brownie-contracts@1.1.1'
dotenv: .env
networks:
  kovan:
    eth_usd_price_feed: '0x9326BFA02ADD2366b30bacB125260Af641031331'
    verify: True
  mainnet-fork:
    eth_usd_price_feed: '0x5f4eC3Df9cbd43714FE2740f5E3616155c5b8419'
    verify: False
  mainnet-fork-dev:
    eth_usd_price_feed: '0x5f4eC3Df9cbd43714FE2740f5E3616155c5b8419'
    verify: False
  development:
    verify: False
  ganache-testnet-dev:
    verify: False
wallets:
  from_key: ${PRIVATE_KEY}
```

I added the **mainnet-fork-dev** with the line ```brownie networks add development mainnet-fork-dev cmd=ganache-cli host=http://127.0.0.1 fork=https://eth-mainnet.alchemyapi.io/v2/API-KEY accounts=10 mnemonic=brownie port=8546```

However, when I run the deploy.py file with the line ```brownie run scripts/deploy.py --network mainnet-fork-dev``` it throws the next:
```
INFORMACIÓN: no se pudo encontrar ningún archivo para los patrones dados.
Brownie v1.16.4 - Python development framework for Ethereum

FundMeBrownieProject is the active project.

Launching 'ganache-cli.cmd --accounts 10 --fork https://eth-mainnet.alchemyapi.io/v2/API-KEY --mnemonic brownie --port 8546 --hardfork istanbul'...

Running 'scripts\deploy.py::main'...
Current network: mainnet-fork-dev
0x66aB6D9362d4F35596279692F0251Db635165871 is deploying a FundMe 
contract
  File "C:\Users\carma\AppData\Local\Programs\Python\Python310\lib\site-packages\brownie\_cli\run.py", line 49, in main
    return_value, frame = run(
  File "C:\Users\carma\AppData\Local\Programs\Python\Python310\lib\site-packages\brownie\project\scripts.py", line 103, in run    
    return_value = f_locals[method_name](*args, **kwargs)        
  File ".\scripts\deploy.py", line 28, in main
    deploy_fund_me()
  File ".\scripts\deploy.py", line 19, in deploy_fund_me
    fund_me=FundMe.deploy(
  File "C:\Users\carma\AppData\Local\Programs\Python\Python310\lib\site-packages\brownie\network\contract.py", line 600, in __call__
    return tx["from"].deploy(
  File "C:\Users\carma\AppData\Local\Programs\Python\Python310\lib\site-packages\brownie\network\account.py", line 507, in deploy 
    data = contract.deploy.encode_input(*args)
  File "C:\Users\carma\AppData\Local\Programs\Python\Python310\lib\site-packages\brownie\network\contract.py", line 629, in encode_input
    data = format_input(self.abi, args)
  File "C:\Users\carma\AppData\Local\Programs\Python\Python310\lib\site-packages\brownie\convert\normalize.py", line 20, in format_input
    raise type(e)(f"{abi['name']} {e}") from None
ValueError: constructor '('0x5f4eC3Df9cbd43714FE2740f5E3616155c5b8419',)' - '('0x5f4eC3Df9cbd43714FE2740f5E3616155c5b8419',)' is not a valid ETH address
Terminating local RPC client...
```

The contract **0x5f4eC3Df9cbd43714FE2740f5E3616155c5b8419** exists in the mainnet. What am I doing wrong?

**Comment by cromewar** [comment]
Hello @carlitox477 for me it seems you have something missing with your constructor which is not taking the correct imput, may be is using this address `0x5f4eC3Df9cbd43714FE2740f5E3616155c5b8419` is it was the account address rather than the price_feed. Remember you must use positional arguments on Python so the position on the deploy matters a lot. I have some recomendations for you.

1. check if your constructor is  correct on the contract.
```SOL
    constructor(address _priceFeed) public {
        priceFeed = AggregatorV3Interface(_priceFeed);
        owner = msg.sender;
    }

```
2. Check if your deploy scripts is correct, remember the position of this two elements `price_feed_address`, `{"from": account}` on the deploy.

```Python
def deploy_fund_me():
    account = get_account()
    if network.show_active() not in LOCAL_BLOCKCHAIN_ENVIRONMENTS:
        price_feed_address = config["networks"][network.show_active()][
            "eth_usd_price_feed"
        ]
    else:
        deploy_mocks()
        price_feed_address = MockV3Aggregator[-1].address

    fund_me = FundMe.deploy(
        price_feed_address,
        {"from": account},
        publish_source=config["networks"][network.show_active()].get("verify"),
    )
    print(f"Contract deployed to {fund_me.address}")
    return fund_me
```

3. Don't forget to add the network to your helpful scripts (and import it on the deploy script)

```Python
FORKED_LOCAL_ENVIRONMENTS = ["mainnet-fork", "mainnet-fork-dev"]
LOCAL_BLOCKCHAIN_ENVIRONMENTS = ["development", "ganache-local"]
```

Finally make sure your alchemy project is on development.
![image](https://user-images.githubusercontent.com/20199180/140061741-61978239-3281-4bab-b45c-7c0cc39b77b4.png)

**Reply by carlitox477** [reply]
Thanks for taking the time to read my problem and answer, but I solved it. If you wan't to laught read the answer below XD

**Reply by cromewar** [reply]
Hahah sometimes we struggle with this kind of errors for hours, I'm glad you solved it haha.

**Comment by carlitox477** [comment, marked as answer]
I found the answer to my problem after spending 4 hours looking the bug: I'm dumb XD

It was a sintax error. Instead of ```price_feed_address=config["networks"][network.show_active()]["eth_usd_price_feed"]``` I put ```price_feed_address=config["networks"][network.show_active()]["eth_usd_price_feed"],```. If you don't see the problem like me, here a hint: look for the comma (**,**) .

So, that's why price_feed_address was interpreted as a tuple and that's why I got the error: 
```
ValueError: constructor '('0x5f4eC3Df9cbd43714FE2740f5E3616155c5b8419',)' - '('0x5f4eC3Df9cbd43714FE2740f5E3616155c5b8419',)' is not a valid ETH address
```

**Reply by cromewar** [reply]
Jum I never would have seen that comma, I'm glad you solved it.

**Reply by GitGAMERGLOVES** [reply]
So mine wouldn't work with the (,) but when removing it - worked!

**Reply by IamSiddharthChoudhary** [reply]
I had the same error but a different syntax issue, and spent 2 days trying out different things. But with this post I fortunately solved the error, So a big thanks from my side.

**Reply by Mobey-eth** [reply]
[IamSiddharthChoudhary](https://github.com/IamSiddharthChoudhary) what was your syntax issiue please
